optaplanner-core/src/main/java/org/optaplanner/core/api/score/buildin/bendable/BendableScore.java

```diff
@@ -105,6 +105,14 @@ protected BendableScore(int initScore, int[] hardScores, int[] softScores) {
         this.softScores = softScores;
     }
 
+    public int[] getHardScores() {
+        return Arrays.copyOf(hardScores, hardScores.length);
+    }
+
+    public int[] getSoftScores() {
+        return Arrays.copyOf(softScores, softScores.length);
+    }
+
     // Intentionally no getters for the hardScores or softScores int arrays to guarantee that this class is immutable
```

Comment:
Can we remove this comment (actually put it above the Arrays.copyOf() and change it to explain why it's a copyOf).